tests/Microsoft.Windows.Setting.Accessibility/Microsoft.Windows.Setting.Accessibility.Tests.ps1

```diff
@@ -1,141 +1,161 @@
-        Invoke-DscResource -Name MousePointer -ModuleName Microsoft.Windows.Setting.Accessibility -Method Set -Property $desiredState
-   
-        $finalState = Invoke-DscResource -Name MousePointer -ModuleName Microsoft.Windows.Setting.Accessibility -Method Get -Property @{}
-        $finalState.PointerSize | Should -Be $desiredPointerSize
-    }
-}
-
-Describe 'VisualEffect'{
-    It 'AlwaysShowScrollbars.'{
-        Invoke-DscResource -Name VisualEffect -ModuleName Microsoft.Windows.Setting.Accessibility -Method Set -Property @{ AlwaysShowScrollbars = $false }
-
-        $initialState = Invoke-DscResource -Name VisualEffect -ModuleName Microsoft.Windows.Setting.Accessibility -Method Get -Property @{}
-        $initialState.AlwaysShowScrollbars | Should -Be $false
-
-        # Set 'AlwaysShowScrollbars' to true.
-        $parameters = @{ AlwaysShowScrollbars = $true }
-        $testResult = Invoke-DscResource -Name VisualEffect -ModuleName Microsoft.Windows.Setting.Accessibility -Method Test -Property $parameters
-        $testResult.InDesiredState | Should -Be $false
-
-        # Verify the changes are correct.
-        Invoke-DscResource -Name VisualEffect -ModuleName Microsoft.Windows.Setting.Accessibility -Method Set -Property $parameters
-        $finalState = Invoke-DscResource -Name VisualEffect -ModuleName Microsoft.Windows.Setting.Accessibility -Method Get -Property @{}
-        $finalState.AlwaysShowScrollbars | Should -Be $true
-
-        $testResult2 = Invoke-DscResource -Name VisualEffect -ModuleName Microsoft.Windows.Setting.Accessibility -Method Test -Property $parameters
-        $testResult2.InDesiredState | Should -Be $true
-    }
-}
-
-AfterAll {
-    $env:TestRegistryPath = ""
-}
\ No newline at end of file
+
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License.
+using module Microsoft.Windows.Setting.Accessibility
+
+$ErrorActionPreference = "Stop"
+Set-StrictMode -Version Latest
+
+<#
+.Synopsis
+   Pester tests related to the Microsoft.Windows.Setting.Accessibility PowerShell module.
+#>
+
+BeforeAll {
+    Install-Module -Name PSDesiredStateConfiguration -Force -SkipPublisherCheck
+    Import-Module Microsoft.Windows.Setting.Accessibility
+
+    # Create test registry path.
+    New-Item -Path TestRegistry:\ -Name TestKey
+    # Set-ItemProperty requires the PSDrive to be in the format 'HKCU:'.
+    $env:TestRegistryPath = ((Get-Item -Path TestRegistry:\).Name).replace("HKEY_CURRENT_USER", "HKCU:")
+}
+
+Describe 'List available DSC resources' {
+    It 'Shows DSC Resources' {
+        $expectedDSCResources = "Text", "Magnifier", "MousePointer", "VisualEffect"
+        $availableDSCResources = (Get-DscResource -Module Microsoft.Windows.Setting.Accessibility).Name
+        $availableDSCResources.length | Should -Be 4
+        $availableDSCResources | Where-Object { $expectedDSCResources -notcontains $_ } | Should -BeNullOrEmpty -ErrorAction Stop
+    }
+}
+
+Describe 'Text' {

```

Comment:
General thoughts on the test design:

- I would consider using [data-driven testing](https://pester.dev/docs/usage/data-driven-tests) to give you a few cases to verify behavior. In general, I would suggest testing the minimal case (key and mandatory properties only) and full case (all compatible properties) where possible. It may also be useful to define cases for resources where you pre-configure the system directly and then use the resource to set it.

  Data-driven tests would also make the test logic easier to read, since it will rely on accessing data sent to the test cases instead of inlining it. It's also much easier to add new test cases in this model for when you define new properties or change behavior.
- In _addition to_ comparing the output from `Invoke-DscResource` after a set, I would strongly recommand a direct check on the settings you care about - otherwise the tests can't verify that the system actually changed, only that the resource reported a change.
- These tests seem to me largely acceptance tests, which is what I always recommend people write first. There's room to add unit tests, but acceptance tests with a few test cases is better than 99% of the resources people author in their own orgs and 80% of the published resources.

Reply:
cc @ryfu-msft

Reply:
Thanks @michaeltlombardi for the great advice on testing. 

@stephengillie, lets leave this comment open for now, I will address this in a separate PR so that it doesn't interfere with your current work.